ach/src/main/java/com/adyen/checkout/ach/DefaultACHDirectDebitDelegate.kt

```diff
@@ -0,0 +1,357 @@
+                encryptedBankLocationId = encryptedBankLocationId,
+                ownerName = outputData.ownerName.value
+            )
+            val paymentComponentData = PaymentComponentData(order = order, paymentMethod = achPaymentMethod)
+
+            if (isAddressRequired(outputData.addressUIState)) {
+                paymentComponentData.billingAddress = AddressFormUtils.makeAddressData(
+                    addressOutputData = outputData.addressState,
+                    addressFormUIState = outputData.addressUIState
+                )
+            }
+
+            return PaymentComponentState(paymentComponentData, isInputValid = true, isReady = true)
+        } catch (e: EncryptionException) {
+            exceptionChannel.trySend(e)
+            return PaymentComponentState(
+                data = PaymentComponentData(),
+                isInputValid = false,
+                isReady = true
+            )
+        }
+    }
+
+    private fun isAddressRequired(addressFormUIState: AddressFormUIState): Boolean {
+        return AddressFormUtils.isAddressRequired(addressFormUIState)
+    }
+
+    override fun observe(
+        lifecycleOwner: LifecycleOwner,
+        coroutineScope: CoroutineScope,
+        callback: (PaymentComponentEvent<PaymentComponentState<ACHDirectDebitPaymentMethod>>) -> Unit
+    ) {
+        observerRepository.addObservers(
+            stateFlow = componentStateFlow,
+            exceptionFlow = exceptionFlow,
+            submitFlow = submitFlow,
+            lifecycleOwner = lifecycleOwner,
+            coroutineScope = coroutineScope,
+            callback = callback
+        )
+    }
+
+    override fun removeObserver() {
+        observerRepository.removeObservers()
+    }
+
+    override fun initialize(coroutineScope: CoroutineScope) {
+        _coroutineScope = coroutineScope
+        submitHandler.initialize(coroutineScope, componentStateFlow)
+
+        sendAnalyticsEvent(coroutineScope)
+        fetchPublicKey(coroutineScope)
+
+        if (componentParams.addressParams is AddressParams.FullAddress) {
+            subscribeToStatesList()
+            subscribeToCountryList()
+            requestCountryList()
+        }
+    }
+
+    override fun setInteractionBlocked(isInteractionBlocked: Boolean) {
+        submitHandler.setInteractionBlocked(isInteractionBlocked)
+    }
+
+    override fun onCleared() {
+        removeObserver()
+        _coroutineScope = null
+    }
+
+    override fun onSubmit() {
+        val state = _componentStateFlow.value
+        submitHandler.onSubmit(
+            state = state
+        )
+    }
+
+    override fun isConfirmationRequired(): Boolean = _viewFlow.value is ButtonComponentViewType
+
+    override fun shouldShowSubmitButton(): Boolean = isConfirmationRequired() && componentParams.isSubmitButtonVisible
+
+    companion object {
+        private val TAG = LogUtil.getTag()
+        const val BANK_ACCOUNT_NUMBER = "bankAccountNumber"
```

Comment:
These should be private

Reply:
Also they're not very descriptive, the name should indicate what exactly are we using these fields for